tests/execution_engine/test_pandas_execution_engine.py

```diff
@@ -766,3 +796,53 @@ def test_get_batch_with_split_on_divided_integer_and_sample_on_list(test_df):
     assert split_df.dataframe.shape == (2, 10)
     assert split_df.dataframe.id.min() == 54
     assert split_df.dataframe.id.max() == 59
+
+
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_constructor_with_azure_options(mock_azure_conn):
+    # default instantiation
+    PandasExecutionEngine()
+
+    # instantiation with custom parameters
+    engine = PandasExecutionEngine(discard_subset_failing_expectations=True)
+    assert "discard_subset_failing_expectations" in engine.config
+    assert engine.config.get("discard_subset_failing_expectations") is True
+    custom_azure_options = {"account_url": "my_account_url"}
+    engine = PandasExecutionEngine(azure_options=custom_azure_options)
+    assert "azure_options" in engine.config
+    assert engine.config.get("azure_options")["account_url"] == "my_account_url"
+
+
+@mock.patch(
+    "great_expectations.execution_engine.pandas_execution_engine.BlobServiceClient",
+)
+def test_get_batch_data_with_azure_batch_spec(
+    mock_azure_conn,
+    azure_batch_spec,
+):
+    mock_blob_client = mock_azure_conn().get_blob_client()
+    mock_azure_obj = mock_blob_client.download_blob()
+    mock_azure_obj.readall.return_value = (
+        b"colA,colB,colC\n1,2,3\n4,5,6\n7,8,9"  # (3,3) CSV for testing
+    )
+
+    df = PandasExecutionEngine().get_batch_data(batch_spec=azure_batch_spec)
+
+    mock_azure_conn().get_blob_client.assert_called_with(
+        container="test_container", blob="path/A-100.csv"
+    )
+    mock_azure_obj.readall.assert_called_once()
+
+    assert df.dataframe.shape == (3, 3)
```

Comment:
We use mocks to abstract away much of the Azure SDK's behavior but ensure that we make the right method calls and end up with a functional `df` at the end.

I'm happy with this since our integration test will place these same methods under more stringent test.

Reply:
Agreed. Thank you for this